### PR TITLE
Add simple browser tests (SHUUP-2698)

### PR DESCRIPTION
**Pull request by Pikkupomo**
Test checkout process in frontend
Test admin login and orders page load

Refs SHUUP-2698

### CODE REVIEW

File: shuup_tests/conftest.py [hunk 1]
```diff
@@ -34,3 +34,6 @@ def pytest_runtest_teardown(item, nextitem):
     if hasattr(item.session, "_theme_overrider"):
         item.session._theme_overrider.__exit__(None, None, None)
         del item.session._theme_overrider
+
+# def pytest_addoption(parser):
```
Comment (tulimaki) [inline shuup_tests/conftest.py]: No need to add commented code

File: shuup_tests/front/test_checkout.py [hunk 2]
```diff
@@ -72,3 +77,102 @@ def test_address_phase_anonymous_user(rf):
...
+def create_orderable_product(name, sku, price):
+    supplier = get_default_supplier()
+    shop = get_default_shop()
+    product = create_product(sku=sku, shop=shop, supplier=supplier, default_price=price, name=name)
+    return product
+
+
+@pytest.mark.browser
+@pytest.mark.djangodb
+def test_browser_checkout(browser, live_server, settings):
```
Comment (tulimaki) [inline shuup_tests/front/test_checkout.py]: I bet you could split this test a little bit and make it more readable.

File: shuup_tests/front/test_checkout.py [hunk 3]
```diff
@@ -72,3 +77,102 @@ def test_address_phase_anonymous_user(rf):
...
+
+    # open product
+    browser.find_by_id("product-1").click()
+
+    # add product to basket
+    browser.find_by_id("add-to-cart-button").click()
+
+    # TODO: make this check better
+    # assume it was added to cart
+    assert browser.is_text_present("(1)")
```
Comment (tulimaki) [inline shuup_tests/front/test_checkout.py]: Yeah, I think this assert needs some tuning if it's for proving that there is one product in the cart.

File: shuup_tests/front/test_checkout.py [hunk 4]
```diff
@@ -72,3 +77,102 @@ def test_address_phase_anonymous_user(rf):
     resp = view_func(request)
     assert 'company' in resp.context_data['form'].form_defs
 
+
+# BROWSER TESTS BELOW
```
Comment (tulimaki) [inline shuup_tests/front/test_checkout.py]: Unneccessary notification about incoming browser tests. you should probably separate the browser tests for checkout in it's own file if you also think that this file needs to be divided.

**Comment by tulimaki** [comment]
Also this commit is missing changelog and documentation.

File: shuup_tests/front/test_checkout.py [hunk 5]
```diff
@@ -72,3 +77,102 @@ def test_address_phase_anonymous_user(rf):
...
+    customer_city = "Test City"
+    customer_country = "US"
+
+    browser.fill("billing-name", customer_name)
+    browser.fill("billing-street", customer_street)
+    browser.fill("billing-city", customer_city)
+    browser.select("billing-country", customer_country)
+    # browser.execute_script('document.getElementById("same_as_billing").checked=true')  # same as billing
+
+    browser.find_by_css(".btn.btn-primary.btn-lg.pull-right").first.click()  # click "continue"
```
Comment (gurch101) [inline shuup_tests/front/test_checkout.py]: selecting by generic bootstrap classes will make these tests brittle imo (plus harder to read without your comments :)). Is it possible to do something similar to this? "#addresses button[type='submit']"

File: shuup_tests/front/test_checkout.py (hunk 2 above)
Comment (Pikkupomo) [inline shuup_tests/front/test_checkout.py]: I will add comments to make it more readable

File: shuup_tests/front/test_checkout.py (hunk 4 above)
Comment (Pikkupomo) [inline shuup_tests/front/test_checkout.py]: I'm not 100% sure I want to divide browser tests into their own files. I was thinking about this, but I think we need to make it later IF the tests are long. This test here is exceptionally long as it proves the whole checkout process.

File: shuup_tests/front/test_checkout.py (hunk 5 above)
Comment (Pikkupomo) [inline shuup_tests/front/test_checkout.py]: Might be possible, not sure. Need to investigate. This is the reason I added the comments :)

**Comment by suutari-ai** [comment]
I think browser tests should be under a separate directory, e.g. `shuup_tests/browser`.

**Comment by suutari-ai** [comment]
Great stuff, @Pikkupomo! And now it seems to work on Travis too (and is skipped by Jenkins).

Let's try this.

**Pull request merged**